packages/packagers/css/src/CSSPackager.js

```diff
@@ -106,77 +134,112 @@ export default (new Packager({
                   }
                 }
 
-                if (media.length) {
-                  return `@media ${media.join(', ')} {\n${css}\n}\n`;
-                }
-
                 return css;
               }),
-              bundle.env.sourceMap && asset.getMapBuffer(),
+              bundle.env.sourceMap ? asset.getMap() : null,
             ]);
           }
         });
       },
     });
 
-    let outputs = await queue.run();
-    let contents = '';
+    let outputs = new Map(
+      (await queue.run()).map(([asset, code, map]) => [asset, [code, map]]),
+    );
     let map = new SourceMap(options.projectRoot);
-    let lineOffset = 0;
 
-    for (let url of hoistedImports) {
-      contents += `@import "${url}";\n`;
-      lineOffset++;
-    }
+    let res = await bundleAsync({
```

Comment:
What can we do for the REPL where `bundleAsync` doesn't exist?

Reply:
It does exist now (as of a few months ago)! Uses emscripten's asyncify transform to turn the rust function into a coroutine.

Reply:
Ah, perfect!

Reply:
We will need to usual aliase to the wasm package though like in the CSS packager and optimizer:
https://github.com/parcel-bundler/parcel/blob/6801029b7ad89fc264ff5554e95ee0fcd6fe699c/packages/optimizers/css/package.json#L34-L36

Reply:
added the dep. are we just relying on the transformer to initialize the wasm module? looks like CSSOptimizer doesn't do it, so I just followed that.

Reply:
Good point, I apparently forgot that in the optimizer. It's not really safe to assume that every worker running the packager also has also at least once invoked the CSS transformer.

Reply:
ok added to both